src/base/bittorrent/session.cpp

```diff
@@ -1199,7 +1199,8 @@ void Session::initializeNativeSession()
         m_nativeSession->add_extension(&lt::create_ut_pex_plugin);
 
 #if (LIBTORRENT_VERSION_NUM < 10200)
-    m_nativeSession->add_extension(boost::shared_ptr<lt::plugin> {new NativeSessionExtension});
+    boost::shared_ptr<lt::plugin> ptr = boost::make_shared<NativeSessionExtension>();
```

Comment:
>Do you know another way to lift the ambiguity for the compiler?

I tried that not long before and didn't found other ways.

I think using `boost::make_shared` isn't valuable here since it need to split to 2 lines (and also longer code) to do the same thing. The `shared_ptr` constructor is quite readable too.

Reply:
what about try to use [boost::static_pointer_cast](https://www.boost.org/doc/libs/1_72_0/libs/smart_ptr/doc/html/smart_ptr.html#pointer_cast) (or maybe boost::dynamic_pointer_cast, I don't know details and what will be suitable in that case), i.e.:
```
m_nativeSession->add_extension(boost::static_pointer_cast<lt::plugin>(oost::make_shared<NativeSessionExtension>()));
```

Reply:
@Chocobo1 but make_shared is safer and more memory efficient that's why it is the preferred way of initializing a shared_ptr

Reply:
>Do you know another way to lift the ambiguity for the compiler?

Just leave it as is. It doesn't seem so important to puzzle over it. It will go into the past with libtorrent-1.1.